less_users/views.py

```diff
@@ -40,104 +45,176 @@ class ProfileView(LoginRequiredMixin, View):
 
 
+def activate_view(request, pk):
+    """it checks if there exists active user_challenge for logged-in user and chosen challenge;
+    if not- it creates new active user_challenge"""
+    user_challenge = UserChallenge.objects.filter(
+        user=request.user, challenge_id=pk, is_active=True, is_deleted=False
+    )
+    if user_challenge.exists():
+        messages.warning(request, "You still have this challenge active!")
+        return redirect("my_challenges")
+    else:
+        UserChallenge.objects.create(user=request.user, challenge_id=pk)
+        messages.success(request, "You activated new challenge!")
+        return HttpResponseRedirect(reverse("challenge_detail", args=[str(pk)]))
+
+
+def event_view_done(request):
+    """gain points when user challenge is completed by clicking >done< button;
+    should check if points can be added (depending on duration and frequency in model)"""
+    try:
+        challenge_id = int(request.POST.get("done"))
+    except ValueError:
+        messages.warning(request, f"This challenge does not exist")
+    else:
+        if Challenge.objects.filter(id=challenge_id).exists():
+            points = Challenge.objects.get(id=challenge_id).points
+            user_challenge = UserChallenge.objects.get(
+                user_id=request.user, challenge_id=challenge_id, is_active=True
+            )
+            user_challenge.check_if_active()
+            if user_challenge.is_active:
+                """proceed user getting points for user challenge; it is up to date"""
+                new_points = user_challenge.get_points()
+                if new_points == points:
+                    Log.objects.create(
+                        user_challenge_id=user_challenge.id, points=points
+                    )
+                    my_profile = request.user.profile
+                    my_profile.points += points
+                    my_profile.save()
+                    messages.success(
+                        request,
+                        f"You have just got {new_points} points for challenge: {user_challenge.challenge}!",
+                    )
+                else:
+                    messages.warning(
+                        request,
+                        f"You cannot get new points for this challenge yet: {user_challenge.challenge}. "
+                        f"You have already done it within last {user_challenge.challenge.frequency} days",
+                    )
+            else:
+                """stop user challenge is out of date"""
+                user_challenge.is_active = False
+                user_challenge.save()
+                messages.warning(
+                    request,
+                    f"You cannot get new points for this challenge: {user_challenge.challenge}. "
+                    f"Its duration time passed",
+                )
+        else:
+            messages.warning(request, f"This challenge does not exist")
```

Comment:
dlaczego zwracasz 200? Patrząc na tą implementację to chyba lepiej by się sprawdzało jako request api a nie widok z renderowaniem html

Reply:
Zróbmy to w osobnej pr